### PR TITLE
Close new windows after downloading completely

### PR DESCRIPTION
**Pull request by fujunwei**
Chrome Browser adds new windows in tab manager that will close after
downloading completely, but NWjs need to close it manually.

BUG=https://github.com/nwjs/nw.js/issues/5734

### CODE REVIEW

File: chrome/browser/download/download_ui_controller.cc [hunk 1]
```diff
@@ -159,6 +162,17 @@ void DownloadUIController::OnDownloadUpdated(content::DownloadManager* manager,
 #if !defined(OS_ANDROID)
   content::WebContents* web_contents = item->GetWebContents();
   if (web_contents) {
```
Comment (rogerwang) [inline chrome/browser/download/download_ui_controller.cc]: What's the URL of this web_contents here?

Comment (fujunwei) [inline chrome/browser/download/download_ui_controller.cc]: It's download url which is http://download.microsoft.com/download/F/7/3/F7395EE6-5642-4AB9-A881-786D0350E88D/Skills_Development_White_Paper_2009.doc based on following html.

 <a target="_blank" href="http://download.microsoft.com/download/F/7/3/F7395EE6-5642-4AB9-A881-786D0350E88D/Skills_Development_White_Paper_2009.doc"> Click Me(Doc) </a>

File: chrome/browser/download/download_ui_controller.cc [hunk 2]
```diff
@@ -159,6 +162,17 @@ void DownloadUIController::OnDownloadUpdated(content::DownloadManager* manager,
...
+    Profile* profile = Profile::FromBrowserContext(web_contents->GetBrowserContext());
+    extensions::AppWindowRegistry* registry = extensions::AppWindowRegistry::Get(profile);
+    if (!registry)
+      return;
+    extensions::AppWindow* app_window = registry->GetAppWindowForWebContents(web_contents);
+    if (!app_window)
+      return;
+    if (app_window->NWCanClose())
```
Comment (fujunwei) [inline chrome/browser/download/download_ui_controller.cc]: Maybe the condition of closing need more strict, will refine it.

Comment (rogerwang) [inline chrome/browser/download/download_ui_controller.cc]: is there any chance that the wrong window is closed?  need to make sure of it.

Comment (fujunwei) [inline chrome/browser/download/download_ui_controller.cc]: Aligned with Chrome browser.

Comment (rogerwang) [inline chrome/browser/download/download_ui_controller.cc]: ok for me to review again? what's the change?

File: chrome/browser/download/download_ui_controller.cc [hunk 3]
```diff
@@ -159,6 +162,20 @@ void DownloadUIController::OnDownloadUpdated(content::DownloadManager* manager,
...
+    Profile* profile = Profile::FromBrowserContext(web_contents->GetBrowserContext());
+    extensions::AppWindowRegistry* registry = extensions::AppWindowRegistry::Get(profile);
+    if (!registry)
+      return;
+    extensions::AppWindow* app_window = registry->GetAppWindowForWebContents(web_contents);
+    if (!app_window)
+      return;
+    if (web_contents->GetController().IsInitialNavigation() &&
+        app_window->NWCanClose() &&
+        !item->IsSavePackageDownload()) {
```
Comment (fujunwei) [inline chrome/browser/download/download_ui_controller.cc]: Please see these changes.

**Comment by rogerwang** [comment]
merged into the master patch. thanks.

**Pull request closed without merging**